sendgrid/exporter.go

```diff
@@ -1,25 +1,42 @@
 package sendgrid
 
 import (
+	"github.com/prometheus/client_golang/prometheus"
 	"log"
 	"time"
-
-	"github.com/prometheus/client_golang/prometheus"
 )
 
 type Exporter struct {
-	client *Client
-	emailLimit      *prometheus.GaugeVec
-	emailRemaining *prometheus.GaugeVec
-	emailUsed       *prometheus.GaugeVec
-	planExpiration  *prometheus.GaugeVec
-	httpReturnCode  *prometheus.GaugeVec
+	client           *Client
+	timeZones        map[string]*time.Location // Map of time locations
+	emailLimit       *prometheus.GaugeVec
+	emailRemaining   *prometheus.GaugeVec
+	emailUsed        *prometheus.GaugeVec
+	planExpiration   *prometheus.GaugeVec
+	httpReturnCode   *prometheus.GaugeVec
 	httpResponseTime *prometheus.GaugeVec
 }
+type AccountConfig struct {
+	AccountName string `yaml:"account_name"`
+	APIKey      string `yaml:"api_key"`
+	TimeZone    string `yaml:"time_zone"`
+}
 
-func NewExporter(apiKeys map[string]string) *Exporter {
+func NewExporter(accounts map[string]AccountConfig) *Exporter {
+	apiKeys := make(map[string]string)
+	timeZones := make(map[string]*time.Location)
+	for accountName, accountConfig := range accounts {
+		apiKeys[accountName] = accountConfig.APIKey
+		loc, err := time.LoadLocation(accountConfig.TimeZone)
+		if err != nil {
+			log.Printf("Error loading time zone for account %s: %v", accountName, err)
+			loc = time.UTC // Default to UTC if time zone cannot be loaded
+		}
+		timeZones[accountName] = loc
+	}
 	return &Exporter{
-		client: NewClient(apiKeys),
+		client:    NewClient(apiKeys), // Pass apiKeys map directly
+		timeZones: timeZones,          // Pass timeZones map
```

Comment:
**chore**: this comment should be removed.

reason: comments should only describe things that are not obvious from reading the code.